src/api/requests/contact.ts

```diff
@@ -1,72 +1,50 @@
 /* eslint-disable @typescript-eslint/no-unused-vars */
-import { timeoutPromise } from '@/utils/timeoutPromise';
 import { PaginationParams } from '../types';
-import { Contact } from '../types/contact';
+import { Contact, ContactMetadata } from '../types/contact';
+import axios from 'axios';
 
 export const searchContacts = async (_pagination?: PaginationParams) => {
-  await timeoutPromise(500);
-  return contacts;
+  const { data: response } = await axios.get(`/contact/search`);
+  return response;
 };
 
-export const addContact = async (paymail: string, name: string) => {
-  await timeoutPromise(1000);
-  contacts = [
-    ...contacts,
-    {
-      paymail,
-      name,
-      status: 'awaiting-acceptance',
-    },
-  ];
+export const upsertContact = async (paymail: string, fullName: string, metadata?: ContactMetadata) => {
+  await axios.put(`/contact/${encodeURIPaymail(paymail)}`, {
+    fullName,
+    metadata,
+  });
 };
 
 export const rejectContact = async (paymail: string) => {
-  console.log('reject');
-  await timeoutPromise(1000);
-  contacts = contacts.filter((contact) => contact.paymail !== paymail);
+  await axios.patch(`/contact/rejected/${encodeURIPaymail(paymail)}`);
 };
 
 export const acceptContact = async (paymail: string) => {
-  await timeoutPromise(1000);
-  contacts = contacts.map((contact) => {
-    if (contact.paymail === paymail) {
-      contact.status = 'not-confirmed';
-    }
-    return { ...contact };
-  });
+  await axios.patch(`/contact/accepted/${encodeURIPaymail(paymail)}`);
 };
 
-export const getTOTP = async (_paymail: string) => {
-  await timeoutPromise(1000);
-  return Math.floor(10 + Math.random() * 89);
+export const getTOTP = async (contact: Contact) => {
+  const res = await axios.post(`/contact/totp`, contact);
+  return res.data.passcode;
 };
 
-export const confirmContactWithTOTP = async (paymail: string, _totp: number) => {
-  await timeoutPromise(1000);
-  contacts = contacts.map((contact) => {
-    if (contact.paymail === paymail) {
-      contact.status = 'confirmed';
-    }
-    return { ...contact };
+export const confirmContactWithTOTP = async (contact: Contact, totp: string) => {
+  await axios.patch(`/contact/confirmed`, {
+    passcode: totp,
+    contact,
   });
 };
 
-/// Mocked contacts
-
-let contacts: Contact[] = [
-  {
-    paymail: 'bob@example.com',
-    name: 'Bob',
-    status: 'confirmed',
-  },
-  {
-    paymail: 'tester@example.com',
-    name: 'Tester',
-    status: 'awaiting-acceptance',
-  },
-  {
-    paymail: 'theguy@example.com',
-    name: 'The Guy',
-    status: 'not-confirmed',
-  },
-];
+const encodeURIPaymail = (paymail: string) => {
+  // Remove control characters from the paymail
+  function* iterator() {
+    for (let i = 0; i < paymail.length; i++) {
+      const code = paymail.charCodeAt(i);
+      if (code > 32 && code !== 127) {
+        yield code;
+      }
+    }
+  }
```

Comment:
Response from AI (I agree with that)

## Option 1: Generator Function

Memory Efficiency: The main advantage of Option 1 is that it uses a generator function with a yield statement. Generators compute and return values on demand, avoiding the creation of a full intermediate array. This means they generally use less memory, especially for larger paymail strings.
No Intermediate Storage: Each character is processed, filtered, and potentially yielded without having to store the entire sanitized string in memory at once.

## Option 2: Array Methods

Intermediate Arrays: Option 2 uses array methods like Array.from(), filter(), and map(). Each of these methods creates a new intermediate array as it transforms the input string. This significantly increases temporary memory usage.
Less Efficient for Large Strings: The larger the input string, the more pronounced the memory difference is likely to be because more intermediate arrays are generated and held in memory before the final result is calculated.
Factors Affecting Memory Consumption

JavaScript Engine Optimizations: Modern JavaScript engines may optimize some array operations in Option 2, reducing memory impact. The actual difference might be less dramatic depending on the engine.
Length of the paymail string: The longer the string, the larger the intermediate arrays in Option 2 become, and therefore the larger the memory difference.
Garbage Collection: How quickly the JavaScript engine's garbage collector can reclaim the memory used by intermediate arrays in Option 2 will impact overall memory usage.
In Summary

For memory efficiency, **Option 1** (the generator function approach) is generally superior, especially when dealing with larger paymail strings. The extent of its advantage can depend on the factors mentioned above.